internal/supported_types/supported_types_interface.go

```diff
@@ -0,0 +1,20 @@
+package supported_types
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+)
+
+type SupportedInstanceTypes struct {
+	Aws struct {
+		Ec2 []string `yaml:"ec2"`
+	} `yaml:"aws"`
+}
+
+var supported SupportedInstanceTypes
```

Comment:
Why these internal structures are exported? I would totally not export those, the only way of getting any results would be IMHO just calling the FilterInstanceTypes function.